core/src/test/scala/unit/kafka/server/ReplicationQuotasTest.scala

```diff
@@ -230,13 +218,31 @@ class ReplicationQuotasTest extends ZooKeeperTestHarness {
 
   private def waitForOffsetsToMatch(offset: Int, partitionId: Int, brokerId: Int): Boolean = {
     waitUntilTrue(() => {
-      offset == brokerFor(brokerId).getLogManager.getLog(TopicAndPartition(topic, partitionId)).map(_.logEndOffset).getOrElse(0)
+      offset == brokerFor(brokerId).getLogManager.getLog(TopicAndPartition(topic, partitionId))
+        .map(_.logEndOffset).getOrElse(0)
     }, s"Offsets did not match for partition $partitionId on broker $brokerId", 60000)
   }
 
   private def property(key: String, value: String) = {
-    new Properties() { put(key, value) }
+    val props = new Properties()
+    props.put(key, value)
+    props
   }
 
   private def brokerFor(id: Int): KafkaServer = brokers.filter(_.config.brokerId == id)(0)
-}
+
+  def createBrokers(brokerIds: Inclusive): Unit = {
+    brokerIds.foreach { id =>
+      brokers = brokers :+ TestUtils.createServer(fromProps(createBrokerConfig(id, zkConnect)))
+    }
+  }
+
+  private def avRate(replicationType: QuotaType, brokers: Inclusive): Double = {
+    brokers.map(brokerFor(_)).map(measuredRate(_, replicationType)).sum / brokers.length
+  }
+
+  private def measuredRate(broker: KafkaServer, repType: QuotaType): Double = {
+    val metricName = broker.metrics.metricName("byte-rate", repType.toString, s"Tracking byte-rate for ${repType}")
```

Comment:
Out of scope for this PR, but it seems like it would be handy to get a metric name without having to pass the description.